### ISSUE TITLE
Setting WordPress config file with post-start hook

### ISSUE THREAD

**Issue by Ebeldev** [drud/ddev#3940]
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Current Behavior

I have set a config.custom.yaml file. 

hooks:
  post-start:
    - exec: "wp config set WP_DEBUG_LOG true"
    - exec: "wp config set WP_DEBUG_DISPLAY false"
    - exec: "wp config set display_errors 0"
    - exec: "wp config set SCRIPT_DEBUG true"
    - exec: "wp config set custom_ENV production"
 

Here is the error. 
Error: Could not process the 'wp-config.php' transformation.
Reason: Unable to locate placement anchor.
Task failed: Exec command 'wp config set SCRIPT_DEBUG true' in container/service 'web': exit status 1

### Expected Behavior

I expect that theses configs settings gets added to the config of my projet.

**Comment by rfay** [comment]
Are you asking for DDEV to do more? Or are you asking why `wp config set SCRIPT_DEBUG true` fails? If you're asking why that fails, is that something that should be filed in the wp-cli issue queue?

**Comment by Ebeldev** [comment]
@rfay Hi Randy, thanks for your reply. ..

Well I just want to set the WP Debug correctly when DDev start. 

As well I wanted to to know if that's the correct way of setting custom constants. 


I'll reach out to wp cli to see if they can see why I cannot set config fails. But I suspect they will tell me to look with DDEV.

**Comment by Ebeldev** [comment]
I did try from ddev ssh

ddev ssh                                                                                                                                                                                                                   
wp config set WP_DEBUG_LOG true
Error: Could not process the 'wp-config.php' transformation.
Reason: Unable to locate placement anchor.

or 

wp config set --raw WP_DEBUG_LOG true
Error: Could not process the 'wp-config.php' transformation.
Reason: Unable to locate placement anchor.

Outside of the container. Just in the terminal where my files are.
I do get the same error. 

wp config set WP_DEBUG_LOG true
Error: Could not process the 'wp-config.php' transformation.
Reason: Unable to locate placement anchor.

or 

wp config set --raw WP_DEBUG_LOG true
Error: Could not process the 'wp-config.php' transformation.
Reason: Unable to locate placement anchor.



From my reseach https://github.com/roots/bedrock/issues/544, it seems that WP CLI does look for the  "/* That's all, stop editing!". And if WP CLI does not found that in th config file it won't set configurations. 

And since DDev does a special config file (wp-config-ddev.php), I do not know if wp cli does understand that when you do use the wp cli it goes in the right file.

**Comment by Ebeldev** [comment]
I just added this to the config file : /* That's all, stop editing! 

and run directly in my terminal not even in ddev ssh.
wp config set WP_DEBUG_LOG true 
And it did work. 

![Screen Shot 2022-06-30 at 12 15 17 AM](https://user-images.githubusercontent.com/8010184/176591520-c745c455-35c3-4bfb-9d3a-aaacdb8ed8f3.png)

If I move the "/* That's all, stop editing! */" in the wp-config-ddev file, then wp cli set config command does not work. 

So what I see is that the wp-config file that is generated by ddev should include this line : "/* That's all, stop editing! */"

Once that runs, then the wp set config command will be able to run in the pre-start hook. 

Otherwise, everytime I ddev start I have to go and reset different config I have. Specially when I'm debugging.

**Comment by Ebeldev** [comment]
But using wp config set with the pre-start hook does not work. 

If the special line is added in the config file then all other config will be written in the config file not in the ddev-config file.

**Comment by Ebeldev** [comment]
I try using the --anchor argument it does work. 

But It would need a special comment like : //customconfigsettings in the wp-config file that is generated by ddev. 

Then use : 
wp config set WP_DEBUG_LOG true --anchor=//customconfigsettings

Then that would work too.

![Screen Shot 2022-06-30 at 12 34 04 AM](https://user-images.githubusercontent.com/8010184/176593523-ffd13552-01af-41a1-adc4-3919a99c7be2.png)

Personnaly, I think it would be better to keep the original string from WordPress config file that wp cli is looking for.

**Comment by rfay** [comment]
@Ebeldev since you seem to completely understand this now, could you please update the original post and the title to reflect what you learned? A PR is welcome, and seems easy. But it does seem like this is an inflexibility in the wp-cli?

**Comment by mandrasch** [comment]
Interesting to learn that WP-CLI looks for `/* That's all, stop editing! Happy publishing. */`. I learn so much here. :-D  :-o 

We discussed in another issue that we don't use the official [wp-config-sample.php](https://github.com/WordPress/WordPress/blob/master/wp-config-sample.php) in DDEV yet, which includes this anchor in line 88 by default:

https://github.com/WordPress/WordPress/blob/master/wp-config-sample.php#L88

By modifying  https://github.com/drud/ddev/blob/a9a72be380e1d850251c61aa1897b4641db5eb23/pkg/ddevapp/wordpress/wp-config.php to match [official wp-config-sample.php](https://github.com/WordPress/WordPress/blob/master/wp-config-sample.php)  we should get rid of the error and be more standard-compliant as well. 

Nice catch @Ebeldev! If my time allows it I could provide help testing a PR! Cheers!

**Comment by rfay** [comment]
I'm fine with a PR to deal with this, it doesn't seem like it's actually something that relates to DDEV at this point, but I'm sure open to contributions. Thanks for sorting it out!

**Comment by Ebeldev** [comment]
@rfay or @mandrasch Was this put in a pr request ? What this solved ?

**Comment by rfay** [comment]
See https://github.com/drud/ddev/issues/3940#issuecomment-1172900001 - We were waiting for a PR from you @Ebeldev ... or anybody :)

**Comment by paulrudy** [comment]
I created a PR #5434 to hopefully address this issue. It's my first PR on this repo. I hope I did it correctly.

Meanwhile, until/unless that PR is accepted, here's a hacky method I'm using in `config.yaml` to insert the needed comment:
```yaml
  post-start:
    - exec-host: grep -q "/\* That's all, stop editing! Happy publishing. \*/" $DDEV_DOCROOT/wp-config.php || (grep -q "/\*\* Absolute path to the WordPress directory. \*/" $DDEV_DOCROOT/wp-config.php && sed -i "" "s/^\/\*\* Absolute path to the WordPress directory. \*\//\/* Add any custom values between this line and the \"stop editing\" line. *\/\n\n\n\n\/* That's all, stop editing! Happy publishing. *\/\n\n&/" $DDEV_DOCROOT/wp-config.php)
    - exec: wp config set <whatever>
```
The script looks for the existence of the needed line `/* That's all, stop editing! Happy publishing. */" in `wp-config.php`. If it doesn't find it, it looks for the line `/** Absolute path to the WordPress directory. */` and adds the following default wordpress block before it:
```php
/* Add any custom values between this line and the "stop editing" line. */



/* That's all, stop editing! Happy publishing. */

```
After that edit's been made, `exec: wp config set` commands are executed successfully.

**Comment by paulrudy** [comment]
just an update, this was fixed with my #5434 and merged in [v1.22.4](https://github.com/ddev/ddev/releases/tag/v1.22.4)

**Comment by rfay** [comment]
Congrats and thanks @paulrudy !
